task_8_1.py

```diff
@@ -0,0 +1,151 @@
+''' Задача 38: Дополнить телефонный справочник возможностью изменения и удаления данных. Пользователь также может ввести имя или фамилию,
+и Вы должны реализовать функционал для изменения и удаления данных.
+'''
+
+def choose_action(phonebook):
+    while True:
+        print('Что вы хотите сделать?')
+        user_choice = input('1 - Импортировать данные\n2 - Найти контакт\n3 - Добавить контакт\n4 - Изменить контакт\n5 - Удалить контакт\n6 - Просмотреть все контакты\n0 - Выйти из приложения\n')
+        print()
+        if user_choice == '1':
+            file_to_add = input('Введите название импортируемого файла: ')
+            import_data(file_to_add, phonebook)
+        elif user_choice == '2':
+            contact_list = read_file_to_dict(phonebook)
+            find_number(contact_list)
+        elif user_choice == '3': add_phone_number(phonebook)
+        elif user_choice == '4': change_phone_number(phonebook)
+        elif user_choice == '5': delete_contact(phonebook)
+        elif user_choice == '6': show_phonebook(phonebook)
```

Comment:
Лучше, все-таки, располагать код более привычным образом
```suggestion
        elif user_choice == '3':
            add_phone_number(phonebook)
        elif user_choice == '4':
            change_phone_number(phonebook)
        elif user_choice == '5':
            delete_contact(phonebook)
        elif user_choice == '6':
            show_phonebook(phonebook)
```